File: Ransom Note.java
//383. Ransom Note LeetCode
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) 
    {
        int count=0;
        char[] mag=magazine.toCharArray();
        for(int i=0;i<ransomNote.length();i++)
        {   char c=ransomNote.charAt(i);
            for(int j=0;j<mag.length;j++)
            {
               if(mag[j]==c) 
               {
                 count++;
                 mag[j]=0;
                   break;
               }
            }
        }
        return count==ransomNote.length()?true:false; 
    }
}

File: Ransom Note.py
#383 RansomNote LeetCode
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        d={}
        for i in magazine:
            if i in d.keys():
                d[i]+=1
            else:
                d[i]=1
        for j in ransomNote:
            if j not in d.keys():
                return False
            d[j]-=1
            if d[j]<0:
                return False
        return True
